pkg/operator/events/recorder.go

```diff
@@ -40,13 +41,18 @@ func GetControllerReferenceForCurrentPod(client corev1client.PodInterface) (*cor
 	if err != nil {
 		return nil, err
 	}
-	ownerRef := metav1.GetControllerOf(pod)
+	replicaSetOwnerRef := metav1.GetControllerOf(pod)
+	rs, err := rsClient.Get(replicaSetOwnerRef.Name, metav1.GetOptions{})
```

Comment:
only do this if the podOwnerRef is of replicaset.

Reply:
good catch, fixed.